paywall/src/middlewares/walletMiddleware.js

```diff
@@ -46,23 +47,43 @@ export default function walletMiddleware({ getState, dispatch }) {
     return callback()
   }
 
-  /**
-   * When an account was changed, we dispatch the corresponding action
-   * The setAccount action will reset other relevant redux state
-   */
-  walletService.on('account.changed', account => {
-    // Let's poll to detect account changes
-    setTimeout(walletService.getAccount.bind(walletService), POLLING_INTERVAL)
-
-    // If the account is actually different
-    if (!getState().account || getState().account.address !== account) {
-      dispatch(
-        setAccount({
-          address: account,
-        })
-      )
+  // account errors are thrown away, except in dev environment
+  pollWithConditions(
+    pollForAccountChange,
+    POLLING_INTERVAL,
+    checkDisableAccountPolling,
+    config.env === 'dev' ? handlePollError : () => {} // log errors in dev only
+  )
+
+  async function pollForAccountChange() {
+    const state = getState()
+    const currentAccount = state.account && state.account.address
+    const newAccounts = await walletService.getAccount() // errors fall through to the polling handler
+    const newAccount = newAccounts && newAccounts[0]
+    if (newAccount === currentAccount) return
```

Comment:
Since the line `await walletService.getAccount()` is async, isn't there a risk that `currentAccount` might be out of date?